pycbc/dq.py

```diff
@@ -199,11 +200,12 @@ def query_flag(ifo, name, start_time, end_time,
                 for rseg in segs:
                     seg_start = rseg[0] + flag['start_pad']
                     seg_end = rseg[1] + flag['end_pad']
-                    flag_segments.append(segment(seg_start, seg_end))
+                    partial.append(segment(seg_start, seg_end))
 
-            # Apply start / end of the veto definer segment
-            send = segmentlist([segment([veto_def['start'], veto_def['end']])])
-            flag_segments = (flag_segments.coalesce() & send)
+                # Apply start / end of the veto definer segment
+                send = segmentlist([segment([flag['start'],
+                                             flag['end']])])
+                flag_segments += (partial.coalesce() & send)
```

Comment:
You *could* coalesce flag_segments here for performance if dealing with lots of flags that are active a lot ... Probably not needed.

Reply:
I remember from reading the segments module documentation that coalesce is only ever required at one point, namely when reading a set of segments from some source file. 
Coalescing is performed automatically at every subsequent stage of segment arithmetic so further calls are redundant.

Reply:
(However I'm not sure if that is true for the `append()` method as used to construct `partial` ..)

Reply:
It's needed for partial to make sure the subsequent operations work unambiguously. It might be possible for the padding to cause two segments to overlap for example.